tests/testthat/test_edit_attribute.R

```diff
@@ -0,0 +1,43 @@
+testthat::context("edit_attribute")
+
+eml_path <- file.path(system.file(package = "datamgmt"), "dummy_eml_w_attributes.xml")
+eml <- EML::read_eml(eml_path)
+eml_key_path <- file.path(system.file(package = "datamgmt"), "key_eml_w_attributes.xml")
+eml_key <- EML::read_eml(eml_key_path)
+testthat::test_that("attributes are updated in correct slots", {
+
+    eml_test <- edit_attribute(eml, 1, 2, attributeDefinition = "test definition", domain = "numericDomain",
+                               measurementScale = "ratio", unit = "dimensionless", numberType = "whole", definition = NA)
+
+    testthat::expect_match(eml_test@dataset@dataTable[[1]]@attributeList@attribute[[2]]@attributeDefinition, "test definition")
+    testthat::expect_equivalent(eml_test@dataset@dataTable[[1]]@attributeList@attribute[[2]]@measurementScale,
```

Comment:
Interesting that you use a eml_key to check your code. Wouldn't have thought to do it that way. I guess it works!

Reply:
That was Dominic's idea -- great workaround where expect_match was failing because the eml change was a new slot type rather than a character string.